identifier/identifier.go

```diff
@@ -66,62 +89,137 @@ type Block struct {
 	// find the licenses in the normalized text and return a list of SPDX IDs
 	// in case of an error, return as much as we have along with an error
-	licenseResults, err := findAllLicensesInNormalizedData(licenseLibrary, normalizedData)
-	if err != nil {
-		return IdentifierResults{}, err
+	if err = findAllLicensesInNormalizedData(identifierResults, licenseLibrary, normalizedData); err != nil {
+		return
 	}
 
-	if err := FromOptions(&licenseResults, options.Enhancements, licenseLibrary); err != nil {
-		return IdentifierResults{}, err
+	if err = FromOptions(identifierResults, options.Enhancements, licenseLibrary); err != nil {
+		return
 	}
 
-	if err := applyMutatorLicenses(licenseLibrary.LicenseMap, &licenseResults); err != nil {
-		return IdentifierResults{}, err
+	if err = applyMutatorLicenses(licenseLibrary.LicenseMap, identifierResults); err != nil {
+		return
 	}
 
+	// TODO: document why we are initializing here
 	if options.OmitBlocks {
-		licenseResults.Blocks = []Block{}
+		identifierResults.Blocks = []Block{}
 	}
 
-	return licenseResults, err
+	return
 }
 
-func IdentifyLicensesInString(input string, options Options, licenseLibrary *licenses.LicenseLibrary) (IdentifierResults, error) {
+func IdentifyLicensesInString(identifierResults *IdentifierResults, input string, options Options, licenseLibrary *licenses.LicenseLibrary) (err error) {
 	// instantiate normalizedData with the input license text
 	normalizedData := normalizer.NormalizationData{
 		OriginalText: input,
 	}
 
 	// normalize the input license text
-	if err := normalizedData.NormalizeText(); err != nil {
-		return IdentifierResults{}, err
+	if err = normalizedData.NormalizeText(); err != nil {
+		return
 	}
 
-	return Identify(options, licenseLibrary, normalizedData)
+	return Identify(identifierResults, options, licenseLibrary, normalizedData)
 }
 
-func IdentifyLicensesInFile(filePath string, options Options, licenseLibrary *licenses.LicenseLibrary) (IdentifierResults, error) {
-	fi, err := os.Stat(filePath)
-	if err != nil {
-		return IdentifierResults{}, err
+func IdentifyLicensesInFile(filePath string, options Options, licenseLibrary *licenses.LicenseLibrary) (identifierResults IdentifierResults, err error) {
+	// Carry filepath used for matches in result set
+	identifierResults = IdentifierResults{}
+	identifierResults.File = filePath
+	identifierResults.Matches = make(map[string][]Match)
+
+	// Verify filepath exists
+	fi, errStat := os.Stat(filePath)
+	if errStat != nil {
+		return identifierResults, errStat
 	}
+	// Only scan files of reasonable sizes
+	// TODO: make the max. size configurable
 	if fi.Size() > 1000000 {
-		Logger.Errorf("file too large (%v > 1000000)", fi.Size()) // log error, but return nil
-		return IdentifierResults{}, nil
+		err = Logger.Errorf("file too large (%v > 1000000)", fi.Size()) // log error, but return nil
```

Comment:
you removed the skipping of large files here

Reply:
good catch